### DISCUSSION TITLE
Using different ssh listen ports by updater possible?

### DISCUSSION THREAD

**Discussion by badummtss** [BassT23/Proxmox#107] [Ideas]
Hey BassT23,
I have the following problem:
In my Proxmox based Infrastructure I use different ssh listening Ports also known as custom ssh Ports for secure my connections which are exposed to the web.

As far as I can see it, the updater only uses the standard  Port 22.

Isn't it possible to change this port for different configurations per VM / LXC?

If yes how can Implement this function?

Best regards

badummtss

**Comment by BassT23** [comment]
Hi @badummtss 

my script don't use a "specific" port.
I only connect with commands like `ssh foo@bar`

So for my understanding, if you can connect from one pve to the other with `ssh pve2` for example, it work ;)

Or, how do you connect between the containers / VMs via terminal?

**Comment by BassT23** [comment]
Hi @badummtss, 
I read a little about custom ports with ssh [here](https://askubuntu.com/questions/673597/ssh-connect-to-host-127-0-0-1-port-2222-connection-refused)

Did you use more than one specific port, or only the same for all?

**Reply by yagler** [reply]
Hi @BassT23,

I don't mean to side-track this discussion thread, (let me know if it should be a new one) but in my case I'm not using a cluster, but when I run "update -up" I get the following error:

_ssh: connect to host <hostname> port 22: Connection refused
scp: Connection closed_

In my case, I have just disabled ssh on port 22 and defined a single different port that allows ssh.

I am just getting Proxmox set up for the first time and I just looked at Proxmox Updater in the last hour so I might be misunderstanding something.

Best Regards,
yagler

**Reply by BassT23** [reply]
Hi @yagler 
nice to see you here ;)
You are absolutely right.

Please let me know, how did you connect via ssh?

Let me say, if port is 222?
Please an example like `ssh node1 uptime`

So I will implement in config file an variable for custom port as soon as possible.

Hopefully you they use these custom port on all you VMs also.
If not, it would be more difficult to code ;)

**Reply by yagler** [reply]
Hi @BassT23,
Here is the way I have connected through ssh if the ssh port was allowed on port 1234:

ssh user1@server1 -p 1234

There might be use cases for multiple ssh options but it would be fine for me to just set 1 variable in a config file.

Thanks,
Yagler

**Reply by BassT23** [reply]
because we only use root user for updating, we don't need the "user@" part ;)

**Reply by BassT23** [reply]
@yagler or @badummtss,
please make an issue with feature-request tag for it ;)
